azafea/event_processors/endless/metrics/v2/cli.py

```diff
@@ -532,3 +541,36 @@ def do_remove_os_info_quotes(config: Config, args: argparse.Namespace) -> None:
         dbsession.commit()
 
     log.info('All done!')
+
+
+def do_remove_empty_location_info(config: Config, args: argparse.Namespace) -> None:
+    db = Db(config.postgresql)
+    log.info('Remove location events with empty info')
+
+    with db as dbsession:
+        query = dbsession.chunked_query(LocationLabel, chunk_size=args.chunk_size).filter(
+            LocationLabel.info == {
+                'id': '',
+                'city': '',
+                'state': '',
+                'street': '',
+                'country': '',
+                'facility': '',
+            })
+        num_records = query.count()
+
+        if num_records == 0:
+            log.info('-> No locations events with empty info in database')
+            return None
+
+        progress(0, num_records)
+
+        for chunk_number, chunk in enumerate(query, start=1):
+            for record in chunk:
+                dbsession.delete(record)
```

Comment:
Won't this fetch every one of the records from the DB in order to delete it? Wouldn't

```
DELETE FROM ... WHERE ...
```

be a better approach than issuing `n / chunk_size` `SELECT`s and `n` `DELETE`s?

Reply:
I agree with the general point that it is not worth spending too much time improving this job which will be run once. I am just concerned that 7 000 `SELECT`s and 35 000 000 `DELETE`s will take long enough that it would be an improvement to write the SQLAlchemy equivalent of:

```sql
DELETE FROM location_label
WHERE info @@ '{"id": "", "city": "", "state": "", "street": "", "country": "", "facility": ""}'
```

and also less code to write? (I would hope this would be better but I am not an expert!)

Reply:
> I am just concerned that 7 000 `SELECT`s and 35 000 000 `DELETE`s will take long enough that it would be an improvement to write the SQLAlchemy equivalent of:

I agree, that would be both easier and faster, that’s my point 2. I usually use chunks (as it’s done everywhere else) and discuss with @adarnimrod, but it’s probably obvious here.

Reply:
I would prefer the plain SQL over the SQLAlchemy version for the improved runtime as well. The progress bar is good to have when we have commands that take a considerable amount of time.

Reply:
> I would prefer the plain SQL over the SQLAlchemy version for the improved runtime as well. The progress bar is good to have when we have commands that take a considerable amount of time.

OK. I think that I can write an SQLAlchemy request that will generate pure SQL and thus have the same runtime, but I’ll use plain SQL if the solution is not obvious.

Reply:
The whole script removing the 3 million lines took 20 seconds, the SQLAlchemy request was already pure SQL launched on the server. Removing the chunks was enough.